Apply RBAC in SSUI

AparnaKarve opened the pull request:
The new attribute ```?attributes=authorization``` in the /api entrypoint (https://github.com/ManageIQ/manageiq/pull/6597) retrieves all product features identifiers that the user is entitled to.
SSUI needs to hide/show the UI elements based on what the user is entitled to do.

https://trello.com/c/Dr2IKoAt/150-5-ssui-enforce-rbac-at-login
https://bugzilla.redhat.com/show_bug.cgi?id=1298393

Fryguy commented on client/app/services/session.service.js:
```diff
@@ -73,5 +74,69 @@
+        return true;
+      } else {
+        return false;
+      }
```
Do you need the whole if/then construct?  Can't this just be...

```js
function entitledForServiceCatalogs(productFeatures) {
  return angular.isDefined(productFeatures.svc_catalog_provision);
}
```

Same goes for the other similar methods below.

Fryguy commented on client/app/services/session.service.js:
```diff
@@ -73,5 +74,69 @@
+
+    function entitledForServices(productFeatures) {
+      var serviceFeature = lodash.find(model.actionFeatures, function(o) {
+        return o.show === true;
```
Does this function need o.show to be a literal true?  Can this just be...

```js
    function entitledForServices(productFeatures) {
      var serviceFeature = lodash.find(model.actionFeatures, function(o) { return o.show; });
      return angular.isDefined(serviceFeature);
    }
```

AparnaKarve replied:
The above function does need ```o.show``` to be a literal true since that in turn will determine if ```serviceFeature``` should be an ```undefined``` or not.  
```serviceFeature``` should stay ```undefined``` if ```o.show``` is not true.

Fryguy commented on client/app/services/session.service.js:
```diff
@@ -73,5 +74,69 @@
+        dashboard: {show: entitledForDashboard(productFeatures)},
+        services: {show: entitledForServices(productFeatures)},
+        requests: {show: entitledForRequests(productFeatures)},
+        marketplace: {show: entitledForServiceCatalogs(productFeatures)}
```
I feel like alignment would look nicer here and in the next function.

h-kataria commented:
:+1:  verified in SSUI.

AparnaKarve commented:
Waiting for UX to provide layout suggestions for a screen where the RBAC settings are applied for a user who is not entitled to view the ```Requests``` section, as shown in the screenshot below -

<img width="1438" alt="screen shot 2016-02-22 at 11 33 05 am" src="https://cloud.githubusercontent.com/assets/1538216/13267899/6c0dd326-da34-11e5-9123-c9f89276981f.png">

AparnaKarve commented:
@dclarizio I was thinking of getting the layout issue (above) addressed in this PR but on second thoughts, can we get this merged and handle it in a separate PR after we hear back from UX?

I have added RBAC for ```Set Ownership``` and have addressed PR feedback as well.

AparnaKarve commented:
@miq-bot remove_label wip

dclarizio commented:
@AparnaKarve please a few tests that validate the allowed feature ids are honored on the JS side.

AparnaKarve commented:
@dclarizio Added a couple tests to set RBAC rules based on feature identifiers.
